src/app/shared/PersonLink.svelte

```diff
@@ -1,16 +1,32 @@
 <script lang="ts">
-  import cx from 'classnames'
+  import cx from "classnames"
   import {deriveProfileDisplay} from "@welshman/app"
+  import {NodeViewWrapper} from "svelte-tiptap"
   import Anchor from "src/partials/Anchor.svelte"
   import {router} from "src/app/util/router"
+  import type {NodeViewProps} from "@tiptap/core"
+  import type {NProfileAttributes} from "nostr-editor"
 
-  export let pubkey
+  export let node: NodeViewProps["node"] = undefined
+  export let pubkey: string
   export let underline = true
 
-  const path = router.at("people").of(pubkey).toString()
-  const display = deriveProfileDisplay(pubkey)
+  $: attrs = node?.attrs as NProfileAttributes
+
+  $: pubkey = pubkey || attrs?.pubkey || ""
+  $: path = router.at("people").of(pubkey).toString()
+  $: display = deriveProfileDisplay(pubkey)
```

Comment:
Avoid too many reactive declarations, the more we can assign with `const` the better for performance. In this case, path will never change if pubkey doesn't, and `deriveProfileDisplay` is already reactive. Unless pubkey changes here, in which case you're doing the right thing.

Reply:
had to change from constant to display the profile, attrs?.pubkey is undefined on the first render